cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -159,6 +159,25 @@ struct format_compiler {
   int8_t subsecond_precision() const { return specifiers.at('f'); }
 };
 
+/**
+ * @brief Specialized function to return the integer value reading up to the specified
+ * bytes or until an invalid character is encountered.
+ *
+ * @param str Beginning of characters to read.
+ * @param bytes Number of bytes in str to read.
+ * @return Integer value of valid characters read and how many bytes were not read.
+ */
+__device__ thrust::pair<int32_t, size_type> parse_int(const char* str, size_type bytes)
```

Comment:
```suggestion
__device__ thrust::pair<int32_t, size_type> parse_int(char const* str, size_type bytes)
```

Reply:
This is the rare instance where (maybe just due to being accustomed to the pattern) it seems like passing bytes by reference and modifying would be cleaner than returning a pair that needs to be unpacked later, but I guess that goes against the `libcudf` style guide?

Reply:
Yes. It was actually doing this before and I switched it here (or maybe a previous PR) to better communicate what is being returned.